enigma-swing/src/main/java/cuchaz/enigma/gui/panels/right/MultiplayerPanel.java

```diff
@@ -0,0 +1,158 @@
+package cuchaz.enigma.gui.panels.right;
+
+import cuchaz.enigma.gui.Gui;
+import cuchaz.enigma.network.packet.MessageC2SPacket;
+import cuchaz.enigma.utils.I18n;
+
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextField;
+import java.awt.BorderLayout;
+import java.awt.event.ActionEvent;
+import java.util.function.Supplier;
+
+public class MultiplayerPanel extends RightPanel {
+    private final JLabel offlineLabel;
+    private final JPanel whenOfflinePanel;
+	private final JPanel whenOnlinePanel;
+	private final JButton sendPendingMessageButton;
+	private final JScrollPane messageScrollPane;
+	private final JTextField pendingMessageBox;
+	private final JLabel usersTitle;
+	private final JLabel messagesTitle;
+	private final JLabel titleCopy;
+
+	private final Supplier<String> offlineTextProvider = () -> I18n.translate("right_panel.multiplayer.offline_text");
+	private final Supplier<String> usersTitleProvider = () -> I18n.translate("right_panel.multiplayer.users_title");
+	private final Supplier<String> messagesTitleProvider = () -> I18n.translate("right_panel.multiplayer.messages_title");
+	private final Supplier<String> sendButtonTextProvider = () -> I18n.translate("right_panel.multiplayer.send");
+
+	private JPanel panel;
+	private boolean offline;
+
+    public MultiplayerPanel(Gui gui) {
```

Comment:
The existing menu to start or connect to a server is called "Collab" instead of "Multiplayer", so this should be renamed for consistency. "Multiplayer" doesn't really apply either, since Enigma doesn't have players in first place :>